### ISSUE TITLE
Create unittest for adas to cherab consistency

### ISSUE THREAD

**Issue by Mateasek** [cherab/openadas#32]
After discovering problems discussed mainly  in #27 , #20 I would like to create unittests to have some kind of consistency check if rate interpolation of adas rates give data with reasonable precision, charge and element notation is consistent and etc.. For now I will include only adf11 files I am dealing with. Would this be also required for other file types?

This is connected to the idea mentioned in #31, which would provide the access to adas file location and would also allow the tests to download the original files if missing.

- [ ] chare notation consistency
- [] adf11 ionisation rate interpolation precision
- [] adf11 recombination rate interpolation precision
- [] adf11 thermal charge exchange rate interpolation precision

**Comment by mattngc** [comment]
I think this is a good idea. It would be great to have some regression tests that can be run to check the interpolators are giving atomic data at least close to published data. You couldn't cover all the data without recreating the entire atomic database. But you could take a few example slices for hydrogen and another main impurity. This would be sensible. But what data are you going to test against and are you sure it can be included? Remember ADAS data can't be redistributed with the repository under the terms of their licence, hence we download it on request. I think what exactly you are testing against needs more thought and better specification.

**Comment by Mateasek** [comment]
I already made 3 unittests for acd, scd and ccd data files [here](https://github.com/Mateasek/cherab_openadas/blob/feature/unittest_adf11_rates/cherab/openadas/tests/test_adf11_rates.py). This is what I used to check that the changes to interpolators in I made PR #33 were consistent with the OpenAdas Data. I don't know anything about copyright of OpenAdas but since you download it to make the repository I thought using it for unittest would be the same.

**Comment by Mateasek** [comment]
Ok after @mattngc clearing my confusion about the idea of the repository, I understand his comment. Still I think the possibility to test data present in the repository (default or user-broadened) should be here.

**Comment by mattngc** [comment]
agreed... I like the idea but I'm not sure what the right way to do it is. I'd like to think about it some more before we rush into this. Maybe @CnlPepper can comment, but he is probably very busy at the moment.

The best way to benchmark would be to take some samples from the official ADAS interpolators in between the knot points and ensure we can get close to those samples.

But really this just amounts to an interpolator test. Viewed in another way, you could just create a dummy curve with the right scale of data, evaluate it in Mathematica / Matlab or something similar and check we get close. I still feel the users should be able to install and change their own data without it breaking the tests. Perhaps we provide a separate path for this kind of testing.
